dynawo/sources/Modeler/DataInterface/PowSyblIIDM/DYNIIDMExtensions.hpp

```diff
@@ -66,23 +78,66 @@ struct IIDMExtensions {
    */
   template<class T>
   static ExtensionDefinition<T> getExtension(const std::string& libPath) {
-    boost::dll::shared_library extensionLibrary(libPath);
+    std::shared_ptr<boost::dll::shared_library> extensionLibrary;
+    std::unique_lock<std::mutex> lock(librariesMutex_);
+    if (libraries_.count(libPath) > 0) {
+      extensionLibrary = libraries_.at(libPath);
+    } else {
+      try {
+        extensionLibrary = std::make_shared<boost::dll::shared_library>(libPath);
+      } catch (const std::exception&) {
+        // No log here as if extension is not defined, it is not defined for all elements and this function is called for
```

Comment:
the sentence is unclear to me

Reply:
What I want to say is that if I put a log here, there will be one log for every element supporting an external IIDM extension because the library we try to load, in practice, is the same